lib/bemxjst/index.js

```diff
@@ -97,6 +97,13 @@ BEMXJST.prototype.compile = function(code) {
   // Concatenate templates with existing ones
   // TODO(indutny): it should be possible to incrementally add templates
   if (this.tree) {
+    self.BEMContext = self.contextConstructor;
+    for (var i = 0; i < out.oninit.length; i++) {
+      // NOTE: oninit has global context instead of BEMXJST
+      var oninit = out.oninit[i];
+      oninit(self, { BEMContext: self.BEMContext });
```

Comment:
Sorry but looks like a bug. Can you add a test with incrementing (or better concatenating) global variable to check quantity (or better order) of `oninit` runs?